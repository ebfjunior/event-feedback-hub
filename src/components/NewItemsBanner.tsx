import { Button } from '@/components/ui/button';

export type NewItemsBannerProps = {
  count: number;
  onApply: () => void;
};

export function NewItemsBanner({ count, onApply }: NewItemsBannerProps) {
  if (count <= 0) return null;
  return (
    <div className="flex items-center justify-between rounded-md border border-cyan-500/50 bg-cyan-50 p-2 text-sm text-cyan-800 dark:bg-cyan-950/40 dark:text-cyan-100">
      <div>{count} new feedback {count === 1 ? 'submission' : 'submissions'} available</div>
      <Button size="sm" variant="secondary" onClick={onApply}>
        Show
      </Button>
    </div>
  );
}
