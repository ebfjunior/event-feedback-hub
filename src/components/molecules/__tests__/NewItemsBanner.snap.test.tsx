import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { NewItemsBanner } from '../NewItemsBanner';

describe('NewItemsBanner (snapshot)', () => {
	it('renders when count > 0 and triggers onApply', () => {
		const onApply = vi.fn();
		const { container, getByText } = render(<NewItemsBanner count={3} onApply={onApply} />);
		expect(container.firstChild).toMatchSnapshot();
		fireEvent.click(getByText('Show'));
		expect(onApply).toHaveBeenCalled();
	});

	it('renders null when count <= 0', () => {
		const { container } = render(<NewItemsBanner count={0} onApply={() => {}} />);
		expect(container.firstChild).toBeNull();
	});
});


