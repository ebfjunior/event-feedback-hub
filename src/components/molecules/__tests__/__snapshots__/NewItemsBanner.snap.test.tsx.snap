// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewItemsBanner (snapshot) > renders when count > 0 and triggers onApply 1`] = `
<div
  class="flex items-center justify-between rounded-md border border-cyan-500/50 bg-cyan-50 p-2 text-sm text-cyan-800 dark:bg-cyan-950/40 dark:text-cyan-100"
>
  <div>
    3
     new feedback 
    submissions
     available
  </div>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-8 rounded-md px-3 text-xs"
  >
    Show
  </button>
</div>
`;
